src/components/Settings/Settings.tsx

```diff
@@ -284,12 +297,18 @@ Settings.Item = function SettingsItem({
     mode,
     description,
 }: SettingsItemProps) {
+    const {renderRightAdornment, showRightAdornmentOnHover} = useSettingsContext();
     return (
         <div className={b('item', {align, mode})}>
             <label className={b('item-heading')}>
-                <span className={b('item-title', {badge: withBadge})}>
-                    {renderTitleComponent(title)}
-                </span>
+                <Flex className={b('item-title-wrapper')} gap={3}>
+                    <span className={b('item-title', {badge: withBadge})}>
+                        {renderTitleComponent(title)}
+                    </span>
+                    <div className={b('item-right-adornment', {hidden: showRightAdornmentOnHover})}>
```

Comment:
Let's not render an empty block if the function `renderRightAdornment` is not specified